### ISSUE TITLE
Json Parsing Error during invalidating tokens

### ISSUE THREAD

**Issue by droidamar007** [microsoftgraph/msgraph-sdk-java#687]
Hello Devs,

I'm trying to invalidate user's tokens using graph api as below - 

```
final HeaderOption headerOption = new HeaderOption("Content-Type", "application/json");
final List<Option> requestOptions = new LinkedList<>();
requestOptions.add(headerOption);
graphClient.users("user-id")
        .revokeSignInSessions()
        .buildRequest(requestOptions)
        .post();
```

SDK - `implementation 'com.microsoft.graph:microsoft-graph:2.5.0'`

> I've set .default scope for ClientCredentialProvider.

I'm getting below error - 

```
CoreHttpProvider[access$000] - 72Error during http request
Throwable detail: com.microsoft.graph.core.ClientException: Error during http request

com.microsoft.graph.core.ClientException: Error during http request
	at com.microsoft.graph.http.CoreHttpProvider.sendRequestInternal(CoreHttpProvider.java:471)
	at com.microsoft.graph.http.CoreHttpProvider.access$000(CoreHttpProvider.java:72)
	at com.microsoft.graph.http.CoreHttpProvider$1.run(CoreHttpProvider.java:171)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:836)
Caused by: com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected a boolean but was BEGIN_OBJECT at line 1 column 2 path $
	at com.google.gson.Gson.fromJson(Gson.java:944)
	at com.google.gson.Gson.fromJson(Gson.java:897)
	at com.google.gson.Gson.fromJson(Gson.java:846)
	at com.google.gson.Gson.fromJson(Gson.java:817)
	at com.microsoft.graph.serializer.DefaultSerializer.deserializeObject(DefaultSerializer.java:82)
	at com.microsoft.graph.http.CoreHttpProvider.handleJsonResponse(CoreHttpProvider.java:532)
	at com.microsoft.graph.http.CoreHttpProvider.sendRequestInternal(CoreHttpProvider.java:448)
	... 5 more
Caused by: java.lang.IllegalStateException: Expected a boolean but was BEGIN_OBJECT at line 1 column 2 path $
	at com.google.gson.stream.JsonReader.nextBoolean(JsonReader.java:854)
	at com.google.gson.internal.bind.TypeAdapters$3.read(TypeAdapters.java:144)
	at com.google.gson.internal.bind.TypeAdapters$3.read(TypeAdapters.java:133)
	at com.google.gson.Gson.fromJson(Gson.java:932)
	... 11 more

```

I've provided the **User.ReadWrite.All, Directory.ReadWrite.All** permission but **Directory.AccessAsUser.All** permission is not listed in the portal.

[https://docs.microsoft.com/en-us/graph/api/user-revokesigninsessions?view=graph-rest-1.0&tabs=java](url)

-I'm stuck in it with 3 days so prompt help is appreciated.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 4662c3f7-8566-09b3-962c-2cfcccfcfc2f
* Version Independent ID: 1cdc7db2-7c2c-9b23-16f7-05e76f588f6c
* Content: [user: revokeSignInSessions - Microsoft Graph v1.0](https://docs.microsoft.com/en-us/graph/api/user-revokesigninsessions?view=graph-rest-1.0&tabs=java)
* Content Source: [api-reference/v1.0/api/user-revokesigninsessions.md](https://github.com/microsoftgraph/microsoft-graph-docs/blob/master/api-reference/v1.0/api/user-revokesigninsessions.md)
* Product: **users**
* Technology: **microsoft-graph**
* GitHub Login: @krbain
* Microsoft Alias: **MSGraphDocsVteam** 
[AB#8419](https://microsoftgraph.visualstudio.com/0985d294-5762-4bc2-a565-161ef349ca3e/_workitems/edit/8419)

**Comment by baywet** [comment]
Hi everyone,
Thanks for reaching out and for your patience.
Because the issue was in the docs repo and not in this SDK repo, I wasn't aware of it.

Can you please: upgrade the sdk to 2.9.0 and core to 1.0.9 (latest at time of writing) and see if you still have the issue?

Can you also try without setting the content-type header please?

**Comment by droidamar007** [comment]
Hi @baywet,

Thanks for your help on this.

I tried updating the SDK repo to mentioned version but no luck. Getting the same error as above.
```
implementation 'com.microsoft.graph:microsoft-graph-core:1.0.9'
implementation 'com.microsoft.graph:microsoft-graph:2.9.0'
```

When I tried without setting the content-type header, I'm getting below stack trace - 

```
Throwable detail: com.microsoft.graph.http.GraphServiceException: Error code: Request_BadRequest
Error message: A supported MIME type could not be found that matches the content type of the response. None of the supported type(s) 'application/json;odata=minimalmetadata;streaming=true, application/json;odata=minimalmetadata;streaming=false, application/json;odata=minimalmetadata, application/json;odata=fullmetadata;streaming=true, application/json;odata=fullmetadata;streaming=false, application/json;odata=fullmetadata, application/json;odata=nometadata;streaming=true, application/json;odata=nometadata;streaming=false, application/json;odata=nometadata, application/json;streaming=true, application/json;streaming=false, application/json;odata=verbose, application/json' matches the content type 'application/octet-stream'.

POST https://graph.microsoft.com/v1.0/users/5b2008cd-2f9a-43dc-94ca-2fbe9ddd2fd0/microsoft.graph.revokeSignInSessions
SdkVersion : graph-java/v2.9.0
Authorization : [PII_REDACTED]


415 : Unsupported Media Type
[...]

[Some information was truncated for brevity, enable debug logging for more details]
com.microsoft.graph.http.GraphServiceException: Error code: Request_BadRequest
Error message: A supported MIME type could not be found that matches the content type of the response. None of the supported type(s) 'application/json;odata=minimalmetadata;streaming=true, application/json;odata=minimalmetadata;streaming=false, application/json;odata=minimalmetadata, application/json;odata=fullmetadata;streaming=true, application/json;odata=fullmetadata;streaming=false, application/json;odata=fullmetadata, application/json;odata=nometadata;streaming=true, application/json;odata=nometadata;streaming=false, application/json;odata=nometadata, application/json;streaming=true, application/json;streaming=false, application/json;odata=verbose, application/json' matches the content type 'application/octet-stream'.

POST https://graph.microsoft.com/v1.0/users/5b2008cd-2f9a-43dc-94ca-2fbe9ddd2fd0/microsoft.graph.revokeSignInSessions
SdkVersion : graph-java/v2.9.0
Authorization : [PII_REDACTED]


415 : Unsupported Media Type
[...]

[Some information was truncated for brevity, enable debug logging for more details]
	at com.microsoft.graph.http.GraphServiceException.createFromConnection(GraphServiceException.java:501)
	at com.microsoft.graph.http.CoreHttpProvider.handleErrorResponse(CoreHttpProvider.java:503)
	at com.microsoft.graph.http.CoreHttpProvider.sendRequestInternal(CoreHttpProvider.java:423)
	at com.microsoft.graph.http.CoreHttpProvider.send(CoreHttpProvider.java:220)
	at com.microsoft.graph.http.CoreHttpProvider.send(CoreHttpProvider.java:200)
	at com.microsoft.graph.http.BaseRequest.send(BaseRequest.java:345)
	at com.microsoft.graph.requests.extensions.UserRevokeSignInSessionsRequest.post(UserRevokeSignInSessionsRequest.java:51)
	at com.vwsa.move.authservice.service.impl.DriverLogoutServiceImpl.driverLogout(DriverLogoutServiceImpl.java:80)
	at com.vwsa.move.authservice.ui.controller.AuthController.driverLogout(AuthController.java:112)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:190)
	at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:138)
	at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:105)
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:878)
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:792)
	at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:87)
	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1040)
	at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:943)
	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1006)
	at org.springframework.web.servlet.FrameworkServlet.doPost(FrameworkServlet.java:909)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:652)
	at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:883)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:733)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:320)
	at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.invoke(FilterSecurityInterceptor.java:126)
	at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.doFilter(FilterSecurityInterceptor.java:90)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
	at org.springframework.security.web.access.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:118)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
	at org.springframework.security.web.session.SessionManagementFilter.doFilter(SessionManagementFilter.java:137)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
	at org.springframework.security.web.authentication.AnonymousAuthenticationFilter.doFilter(AnonymousAuthenticationFilter.java:111)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
	at org.springframework.security.web.servletapi.SecurityContextHolderAwareRequestFilter.doFilter(SecurityContextHolderAwareRequestFilter.java:158)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
	at org.springframework.security.web.savedrequest.RequestCacheAwareFilter.doFilter(RequestCacheAwareFilter.java:63)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
	at org.springframework.security.oauth2.server.resource.web.BearerTokenAuthenticationFilter.doFilterInternal(BearerTokenAuthenticationFilter.java:114)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
	at org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:116)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
	at org.springframework.security.web.header.HeaderWriterFilter.doHeadersAfter(HeaderWriterFilter.java:92)
	at org.springframework.security.web.header.HeaderWriterFilter.doFilterInternal(HeaderWriterFilter.java:77)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
	at org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:105)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
	at org.springframework.security.web.context.request.async.WebAsyncManagerIntegrationFilter.doFilterInternal(WebAsyncManagerIntegrationFilter.java:56)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
	at org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:215)
	at org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:178)
	at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:358)
	at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:271)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:100)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:93)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.springframework.boot.actuate.metrics.web.servlet.WebMvcMetricsFilter.doFilterInternal(WebMvcMetricsFilter.java:93)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:202)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:541)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:139)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:373)
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65)
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:868)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1589)
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.base/java.lang.Thread.run(Thread.java:836)
```

One thing I observed, I'm not able to give **[Directory.AccessAsUser.All](https://stackoverflow.com/questions/45549105/set-the-scope-parameter-for-microsoft-graph)** permission to the application because it is not listed on my azure b2c portal dashboard. I need to check this with our IT Admin department.

**Comment by baywet** [comment]
Hi @droidamar007 
Thanks for trying with the latest version and for the additional details.
I was able to reproduce the issue and it is two folds:

A while ago, a default value of `application/octet-stream` [was added for post requests with empty bodies](https://github.com/microsoftgraph/msgraph-sdk-java/commit/7621b940704178f57a04db87cc1eb795a80ddf58). I don't believe this is correct as the [content type header should not be provided when the body is empty](https://stackoverflow.com/questions/29784398/should-content-type-header-be-present-when-the-message-body-is-empty)
This can be worked-around by manually setting the header like you did, but ideally consumers should not have to do this.

Once that is "fixed" the deserializer is having a hard-time handling the response payload and throws an exception.
```json
{"@odata.context":"https://graph.microsoft.com/v1.0/$metadata#Edm.Null","@odata.null":true}
```

I'll probably end-up adding a custom deserializer to the existing collection to fix that issue. Once this is done, I'm hoping this API call succeeds. I'll keep posting updates here.
